R/redundant_equals_linter.R

```diff
@@ -30,6 +35,11 @@
 #'   linters = redundant_equals_linter()
 #' )
 #'
+#' lint(
+#'   text = "dt[!(x), y]",
```

Comment:
I prefer keeping `!` inside `()` myself:

```suggestion
#' # in data.table semantics, dt[x] is a join, dt[(x)] is a subset
#' lint(
#'   text = "dt[!(x), y]",
```

Reply:
I might also prefer something more practical like `dt[is_tall = FALSE, y]` --> `dt[(!is_tall), y]`.

Reply:
Thanks, Michael. I think it's a good idea to include a comment and also use a more realistic column name.